libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -268,13 +437,16 @@ void AP_CRSF_Telem::update()
 void AP_CRSF_Telem::update_params()
 {
     // handle general parameter requests
-    switch (_request_pending) {
+    switch (_request_pending.frame_type) {
     case AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAM_DEVICE_PING:
         calc_device_info();
         break;
     case AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAMETER_READ:
         calc_parameter();
         break;
+    case AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAM_DEVICE_INFO:
+        calc_device_ping();
```

Comment:
It seems rather odd that we calculate pin under the device-info case and vice-versa.

Reply:
@andyp1per yes it seemed odd to me too but I respected the pre-existing approach

Reply:
It's request-response - you are pinged and you return a device info. If you are handed a device info, not sure why you then need a ping, the ping should have already happened.